db/revision_cache_test.go

```diff
@@ -2442,3 +2443,41 @@ func TestEvictionWhenStaleCVRemoved(t *testing.T) {
 	revValueRevEntry := valRevEntry.Value.(*revCacheValue)
 	assert.Equal(t, "2-abc", revValueRevEntry.revID)
 }
+
+func TestUpdateDeltaRevCacheMemoryStatPanic(t *testing.T) {
```

Comment:
Would be worth adding a comment to this test that describes why we're interleaving two revcache operations to simulate the concurrency bug.

Perhaps also clearer comments for each "concurrent" operation being run (i.e. `// Thread 1: UpdateDelta (start)` `// Thread 2: Remove (start)`